hedy.py

```diff
@@ -1932,31 +1939,56 @@ def play(self, meta, args):
+                    value = process_characters_needing_escape(value)
+                    return variable_name + " = '" + value + "'" + self.add_debug_breakpoint()
         else:
-            if self.is_variable(value, meta.line):  # if the value is a variable, this is a reassign
-                value = self.process_variable(value, meta.line)
-                return variable_name + " = " + value + self.add_debug_breakpoint()
+            if self.is_random(value) or self.is_list(value):
+                exception = self.make_index_error_check_if_list([value])
+                return "    " + exception + str(variable_name) + " = " + value + self.add_debug_breakpoint()
             else:
-                # if the assigned value is not a variable and contains single quotes, escape them
-                value = process_characters_needing_escape(value)
-                return variable_name + " = '" + value + "'" + self.add_debug_breakpoint()
+                if self.is_variable(value, meta.line):  # if the value is a variable, this is a reassign
+                    value = self.process_variable(value, meta.line)
+                    return "    " + str(variable_name) + " = " + value + self.add_debug_breakpoint()
+                else:
+                    # if the assigned value is not a variable and contains single quotes, escape them
+                    value = process_characters_needing_escape(value)
+                    return "    " + str(variable_name) + " = '" + value + "'" + self.add_debug_breakpoint()
 
     def sleep(self, meta, args):
-        if not args:
-            return f"time.sleep(1){self.add_debug_breakpoint()}"
+        if not self.microbit:
+            if not args:
+                return f"time.sleep(1){self.add_debug_breakpoint()}"
+            else:
+                value = f'"{args[0]}"' if self.is_int(args[0]) else args[0]
+                if not self.is_int(args[0]):
+                    self.add_variable_access_location(value, meta.line)
+                index_exception = self.make_index_error_check_if_list(args)
+                ex = make_value_error(Command.sleep, 'suggestion_number', self.language)
+                code = index_exception + \
+                    textwrap.dedent(f"time.sleep(int_with_error({value}, {ex})){self.add_debug_breakpoint()}")
+                return code
         else:
-            value = f'"{args[0]}"' if self.is_int(args[0]) else args[0]
-            if not self.is_int(args[0]):
-                self.add_variable_access_location(value, meta.line)
-            index_exception = self.make_index_error_check_if_list(args)
-            ex = make_value_error(Command.sleep, 'suggestion_number', self.language)
-            code = index_exception + \
-                textwrap.dedent(f"time.sleep(int_with_error({value}, {ex})){self.add_debug_breakpoint()}")
-            return code
+            if not args:
```

Comment:
For this command the microbit code is quite different than the original non-microbit one. In this case, there is not much gain from the fact that the 2 code snippets (the microbit and the non-microbit one) are right next to each other. Have you considered instead of using `if` statements, to subclass e.g. 
```
@v_args(meta=True)
@microbit_hedy_transpiler(level=2)
@source_map_transformer(source_map)
class MicrobitConvertToPython_2(ConvertToPython_2):
   def sleep(self, meta, args):
      ...
```
In this way the microbit code will be together and the code will not have if-microbit-else statements. However, the downside is that the microbit code might be overlooked when changes are made to the main transpiler.

If you have considered this option and rejected it, please feel free to close this comment. If you haven't and you want some help to explore it further, do not hesitate to ping me.

Reply:
As we discussed this will be the new approach with a separate transpiler for microbit.